docs/user/sbt.rst

```diff
@@ -110,27 +118,49 @@ the options directly is now deprecated.
-===== ======================== ================ =========================================================
+To append a value to the right of any previous setting:
+
+.. code-block:: scala
+
+    import scala.scalanative.build._
+
+    // Enable verbose reporting during compilation
+    nativeConfig ~= { c =>
+      c.withCompileOptions(c.compileOptions ++ Seq("-v"))
+    }
+
+    // Use an alternate linker
+    nativeConfig ~= { c =>
+      c.withLinkingOptions(c.linkingOptions ++ Seq("-fuse-ld=mold"))
+    }
+
+The "Since" column in Table 1 gives the Scala Native version where a concept was introduced.
+The "Name" column gives the name of the current method.
+For prior names and practice, see the documentation for the indicated Scala Native version.
+
+===== ============================ ================  ================================================================
+Table 1
+---------------------------------- ----------------  ----------------------------------------------------------------
+Since Name                         Type              Description
+===== ============================ ================  ================================================================
+0.1   ``withCompile``              ``Analysis``      Compile Scala code to NIR
+0.1   ``withRun``                  ``Unit``          Compile, link and run the generated binary
+0.1   ``withPackage``              ``File``          Similar to standard package with addition of NIR
+0.1   ``withPublish``              ``Unit``          Similar to standard publish with addition of NIR (1)
+0.1   ``withNativeLink``           ``File``          Link NIR and generate native binary
+0.1   ``withNativeClang``          ``File``          Path to ``clang`` command
+0.1   ``withNativeClangPP``        ``File``          Path to ``clang++`` command
+0.1   ``withNativeCompileOptions`` ``Seq[String]``   Extra options passed to clang verbatim during compilation
+0.1   ``withNativeLinkingOptions`` ``Seq[String]``   Extra options passed to clang verbatim during linking
+0.1   ``withNativeMode``           ``String``        One of ``"debug"``, ``"release-fast"`` or ``"release-full"`` (2)
+0.2   ``withNativeGC``             ``String``        One of ``"none"``, ``"boehm"``, ``"immix"`` or ``"commix"`` (3)
+0.3.3 ``withNativeLinkStubs``      ``Boolean``       Whether to link ``@stub`` definitions, or to ignore them
+0.4.0 ``withNativeConfig``         ``NativeConfig``  Configuration of the Scala Native plugin
+0.4.0 ``withNativeLTO``            ``String``        One of ``"none"``, ``"full"`` or ``"thin"`` (4)
+0.4.0 ``withTargetTriple``         ``String``        The platform LLVM target triple
+0.4.0 ``withNativeCheck``          ``Boolean``       Shall the linker check intermediate results for correctness?
+0.4.0 ``withNativeDump``           ``Boolean``       Shall the linker dump intermediate results to disk?
```

Comment:
This new table is misleading and partially incorrect. The first 4 entries are standard sbt commands: `compile`, `run`, `package`, `publish` which under the Scala Native plugin might behave differently. 
All the next entries are defining sbt settings, which are treated as deprecated, they will be removed in the future. `All the `withX` commands are a part of the NativeConfig. We should restore original table and create a separate table describing NativeConfig available configs available for the users (it should contain entries since `withNativeClang` with exception of `withNativeConfig` and should contains new accessors added after 0.4.0, eg. withCheckFatalWarnings, withEmbedResources, withLinktimeProperties)

Reply:
If Fate is kind to me, I will re-work this as suggested.  Thank you for the review.